cleanlab/multiannotator.py

```diff
@@ -0,0 +1,962 @@
+        likelihood = np.mean(
+            annotator_agreement[num_annotations != 1]
+        )  # likelihood that any annotator will annotate the consensus label for any example
+
+        # compute adjusted annotator agreement (used as annotator weights)
+        annotator_agreement_with_annotators = _get_annotator_agreement_with_annotators(
+            labels_multiannotator, num_annotations
+        )
+        annotator_error = 1 - annotator_agreement_with_annotators
+        most_likely_class_error = np.mean(
+            consensus_label != np.argmax(np.bincount(consensus_label, minlength=num_classes))
+        )
+        adjusted_annotator_agreement = np.clip(
+            1 - (annotator_error / most_likely_class_error), a_min=0, a_max=None
+        )
+
+        # compute model weight
+        mask = num_annotations != 1
+        consensus_label_subset = consensus_label[mask]
+        prior_pred_probs_subset = prior_pred_probs[mask]
+
+        model_error = np.mean(np.argmax(prior_pred_probs_subset, axis=1) != consensus_label_subset)
+        model_weight = (1 - (model_error / most_likely_class_error)) * np.sqrt(
+            np.mean(num_annotations)
+        )
+
+        # compute weighted average
+        post_pred_probs = np.full(prior_pred_probs.shape, np.nan)
+        for i in range(len(labels_multiannotator)):
+            example = labels_multiannotator.iloc[i].dropna()
+            post_pred_probs[i] = [
+                np.average(
+                    [prior_pred_probs[i, true_label]]
+                    + [
+                        likelihood
+                        if annotator_label == true_label
+                        else (1 - likelihood) / (num_classes - 1)
+                        for annotator_label in example
+                    ],
+                    weights=np.concatenate(
+                        ([model_weight], adjusted_annotator_agreement[example.index])
+                    ),
+                )
+                for true_label in range(num_classes)
+            ]
+
+        return_model_weight = model_weight
+        return_annotator_weight = adjusted_annotator_agreement
+
+    elif quality_method == "agreement":
+        label_counts = labels_multiannotator.apply(lambda s: s.value_counts(), axis=1)
+        label_counts = np.nan_to_num(label_counts, nan=0)
+        post_pred_probs = label_counts / num_annotations.reshape(-1, 1)
+
+    else:
+        raise ValueError(
+            f"""
+            {quality_method} is not a valid quality method!
+            Please choose a valid quality_method: {valid_methods}
+            """
+        )
+
+    return post_pred_probs, return_model_weight, return_annotator_weight
+
+
+def _get_quality_of_consensus(
+    labels_multiannotator: pd.DataFrame,
+    consensus_label: np.ndarray,
+    pred_probs: np.ndarray,
+    num_annotations: np.ndarray,
+    annotator_agreement: np.ndarray,
+    quality_method: str = "auto",
+    kwargs: dict = {},
```

Comment:
this should be named `XXX_kwargs` where XXX provides extra info about how these kwargs will be used.  This avoids confusion with the standard `**kwargs`